dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java

```diff
@@ -399,266 +419,108 @@ public void list(String processName, PrintStream out, PrintStream err, Integer .
         }
     }
 
-    /**
-     * Register DOI with the provider
-     * @param doiRow        - doi to register
-     * @param filter        - logical item filter to override
-     * @throws SQLException
-     * @throws DOIIdentifierException
-     */
-    public void register(DOI doiRow, Filter filter) throws SQLException, DOIIdentifierException {
+    private void performDOIOperation(DOI doiRow, DOIOperation operation, String operationName) {
```

Comment:
`operationName` could be set as a property of `DOIOperation`, instead of passing it as an argument to this function.
This would allow a much cleaner approach and would bound that specific `operationName` to a given `Strategy` implemented with a specialized `DOIOperation`.